gog/graph.go

```diff
@@ -47,6 +47,16 @@ type Grapher struct {
 
 	seenDocObjs map[types.Object]struct{}
 	seenDocKeys map[string]struct{}
+
+	// pkgDefs accumulates the defs from the package being graphed currently in Graph().
+	// If the package is graphed successfully, the defs are added to Output.
+	pkgDefs []*Def
```

Comment:
Why are these necessary? Why not just append directly to `.Defs`, `.Refs`, and `.Docs`?